Client/Frontend/Widgets/BlurTableViewController.swift

```diff
@@ -0,0 +1,182 @@
+        tableView.snp_makeConstraints { make in
+            make.center.equalTo(self.view)
+            make.width.equalTo(290)
+            make.height.equalTo(73 + actions.count * 56)
+        }
+
+        tableView.delegate = self
+        tableView.dataSource = self
+        tableView.keyboardDismissMode = UIScrollViewKeyboardDismissMode.OnDrag
+        tableView.backgroundColor = UIConstants.PanelBackgroundColor
+        tableView.scrollEnabled = false
+        tableView.separatorColor = UIConstants.SeparatorColor
+        tableView.layer.cornerRadius = 10
+        tableView.registerClass(BlurTableViewCell.self, forCellReuseIdentifier: "BlurTableViewCell")
+        tableView.registerClass(BlurTableViewHeaderCell.self, forCellReuseIdentifier: "BlurTableViewHeaderCell")
+
+        if #available(iOS 9, *) {
+            tableView.cellLayoutMarginsFollowReadableWidth = false
+        }
+    }
+
+    func dismiss(gestureRecognizer: UIGestureRecognizer) {
+        self.dismissViewControllerAnimated(true, completion: nil)
+    }
+
+    deinit {
+        // The view might outlive this view controller thanks to animations;
+        // explicitly nil out its references to us to avoid crashes. Bug 1218826.
+        tableView.dataSource = nil
+        tableView.delegate = nil
+    }
+
+    func numberOfSectionsInTableView(tableView: UITableView) -> Int {
+        return 1
+    }
+
+    func tableView(tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return actions.count + 1
+    }
+
+    func tableView(tableView: UITableView, didSelectRowAtIndexPath indexPath: NSIndexPath) {
+        if indexPath.row == 0 {
+            return
+        }
+        let action = actions[indexPath.row - 1]
+        return action.handler(action)
+    }
+
+    func tableView(tableView: UITableView, hasFullWidthSeparatorForRowAtIndexPath indexPath: NSIndexPath) -> Bool {
+        return true
+    }
+
+    func tableView(tableView: UITableView, heightForRowAtIndexPath indexPath: NSIndexPath) -> CGFloat {
+        return indexPath.row == 0 ? 74 : 56
+    }
+
+    func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
+        if indexPath.row == 0 {
+            let cell = tableView.dequeueReusableCellWithIdentifier("BlurTableViewHeaderCell", forIndexPath: indexPath) as! BlurTableViewHeaderCell
+            cell.preservesSuperviewLayoutMargins = false
+            cell.separatorInset = UIEdgeInsetsZero
+            cell.layoutMargins = UIEdgeInsetsZero
+            cell.configureWithSite(site)
+            return cell
+        }
+
+        let cell = tableView.dequeueReusableCellWithIdentifier("BlurTableViewCell", forIndexPath: indexPath) as! BlurTableViewCell
+        let action = actions[indexPath.row - 1]
+        cell.configureCell(action.title, imageString: action.iconString)
+        return cell
+    }
+}
+
+class BlurTableViewHeaderCell: SimpleHighlightCell {
```

Comment:
You have this class here. But the Other TableViewCell is in its own file. I'd put them all in one file. Also it would be great if this had a more descriptive name. Because it functions very similar to UIAlertController. Some variation of that would make it clear to everyone what it did. I don't think calling it tableview here matters as the implementation of the table is completely abstracted away to someone using this class.